common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java

```diff
@@ -87,13 +94,34 @@ private void buildSchemaRegistryConfigs(Map<String, ?> configs) {
         validateAndSetJacksonDeserializationFeatures(configs);
         validateAndSetTags(configs);
         validateAndSetMetadata(configs);
+        validateAndSetUserAgent(configs);
+        validateAndSetSecondaryDeserializer(configs);
+    }
+
+    private void validateAndSetSecondaryDeserializer(Map<String, ?> configs) {
+        if (isPresent(configs, AWSSchemaRegistryConstants.SECONDARY_DESERIALIZER)) {
+            Object secondaryDeserializer = configs.get(AWSSchemaRegistryConstants.SECONDARY_DESERIALIZER);
+            if (secondaryDeserializer instanceof String) {
+                this.secondaryDeserializer = (String) secondaryDeserializer;
+            } else if (secondaryDeserializer instanceof Class) {
+                this.secondaryDeserializer = ((Class) secondaryDeserializer).getName();
+            } else {
+                throw new AWSSchemaRegistryException("Invalid secondary de-serializer configuration");
+            }
+        }
     }
 
     private void buildCacheConfigs(Map<String, ?> configs) {
         validateAndSetCacheSize(configs);
         validateAndSetCacheTTL(configs);
     }
 
+    private void validateAndSetUserAgent(Map<String, ?> configs) {
+        if (isPresent(configs, AWSSchemaRegistryConstants.USER_AGENT_APP)) {
+            this.userAgentApp = (String) configs.get(AWSSchemaRegistryConstants.USER_AGENT_APP);
```

Comment:
``String.valueOf`` is better to avoid any casting exceptions.

Reply:
> `String.valueOf` is better to avoid any casting exceptions.

I think it's good to fail this during validation if the UserAgentApp value is not set as String to avoid side-effects like setting this to Object hash code.

Reply:
String.valueOf would also be consistent with rest of the class, but i guess this works too.